### PR TITLE
Added Slack Message Attachments

### PR DESCRIPTION
**Pull request by RLittlesII**
PR related to discussion #3

### CODE REVIEW

File: src/Cake.Slack/Chat/SlackChatMessageAttachment.cs [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+        }
+
+        /// <summary>
+        /// Required Text summary of the attachment that is shown by clients that understand attachments but choose not to show them.
+        /// </summary>
+        /// <value>
+        /// The fallback.
+        /// </value>
+        public string Fallback { get; set; }
+        /// <summary>
```
Comment (devlead) [inline src/Cake.Slack/Chat/SlackChatMessageAttachment.cs]: Add new line between previous property and summary

File: src/Cake.Slack/Chat/SlackChatMessageAttachment.cs [hunk 2]
```diff
@@ -0,0 +1,138 @@
...
+        /// Optional Text that should appear within the attachment
+        /// </summary>
+        /// <value>
+        /// The Text.
+        /// </value>
+        public string Text { get; set; }
+        /// <summary>
```
Comment (devlead) [inline src/Cake.Slack/Chat/SlackChatMessageAttachment.cs]: Add new line between previous property and summary

File: src/Cake.Slack/Chat/SlackChatMessageAttachment.cs [hunk 3]
```diff
@@ -0,0 +1,138 @@
...
+        /// Optional Text that should appear above the formatted data
+        /// </summary>
+        /// <value>
+        /// The Pretext.
+        /// </value>
+        public string Pretext { get; set; }
```
Comment (devlead) [inline src/Cake.Slack/Chat/SlackChatMessageAttachment.cs]: Add new line between previous property and summary

File: src/Cake.Slack/Chat/SlackChatMessageAttachment.cs [hunk 4]
```diff
@@ -0,0 +1,138 @@
...
+        /// <summary>
+        /// Color displayed with the attachment
+        /// </summary>
+        /// <value>
+        /// The Color.
+        /// </value>
+        public string Color { get ; set; }
+        /// <summary>
```
Comment (devlead) [inline src/Cake.Slack/Chat/SlackChatMessageAttachment.cs]: Add new line between previous property and summary

File: src/Cake.Slack/Chat/SlackChatMessageAttachment.cs [hunk 5]
```diff
@@ -0,0 +1,138 @@
...
+        /// The message title.
+        /// </summary>
+        /// <value>
+        /// The title.
+        /// </value>
+        public string Title { get; set; }
+        /// <summary>
```
Comment (devlead) [inline src/Cake.Slack/Chat/SlackChatMessageAttachment.cs]: Add new line between previous property and summary

File: src/Cake.Slack/Chat/SlackChatMessageAttachment.cs [hunk 6]
```diff
@@ -0,0 +1,138 @@
...
+        /// The message title link.
+        /// </summary>
+        /// <value>
+        /// The title link.
+        /// </value>
+        public string Title_Link { get; set; }
+        /// <summary>
```
Comment (devlead) [inline src/Cake.Slack/Chat/SlackChatMessageAttachment.cs]: Add new line between previous property and summary

File: src/Cake.Slack/Chat/SlackChatMessageAttachment.cs [hunk 7]
```diff
@@ -0,0 +1,138 @@
...
+        /// The message author's name.
+        /// </summary>
+        /// <value>
+        /// The author_name.
+        /// </value>
+        public string Author_Name { get; set; }
+        /// <summary>
```
Comment (devlead) [inline src/Cake.Slack/Chat/SlackChatMessageAttachment.cs]: Add new line between previous property and summary

File: src/Cake.Slack/Chat/SlackChatMessageAttachment.cs [hunk 8]
```diff
@@ -0,0 +1,138 @@
...
+        /// Link to the message author
+        /// </summary>
+        /// <value>
+        /// The author_link.
+        /// </value>
+        public string Author_Link { get; set; }
+        /// <summary>
```
Comment (devlead) [inline src/Cake.Slack/Chat/SlackChatMessageAttachment.cs]: Add new line between previous property and summary

File: src/Cake.Slack/Chat/SlackChatMessageAttachment.cs [hunk 9]
```diff
@@ -0,0 +1,138 @@
...
+        /// The message author's icon url
+        /// </summary>
+        /// <value>
+        /// The author_icon.
+        /// </value>
+        public string Author_Icon { get; set; }
+        /// <summary>
```
Comment (devlead) [inline src/Cake.Slack/Chat/SlackChatMessageAttachment.cs]: Add new line between previous property and summary

File: src/Cake.Slack/Chat/SlackChatMessageAttachment.cs [hunk 10]
```diff
@@ -0,0 +1,138 @@
...
+        /// Url to an image to display in message.
+        /// </summary>
+        /// <value>
+        /// The image_url.
+        /// </value>
+        public string Image_Url { get; set; }
+        /// <summary>
```
Comment (devlead) [inline src/Cake.Slack/Chat/SlackChatMessageAttachment.cs]: Add new line between previous property and summary

File: src/Cake.Slack/Chat/SlackChatMessageAttachment.cs [hunk 11]
```diff
@@ -0,0 +1,138 @@
...
+        /// Url to thumbprint to display in message.
+        /// </summary>
+        /// <value>
+        /// The thumb_url.
+        /// </value>
+        public string Thumb_Url { get; set; }
+        /// <summary>
```
Comment (devlead) [inline src/Cake.Slack/Chat/SlackChatMessageAttachment.cs]: Add new line between previous property and summary

File: src/Cake.Slack/Chat/SlackChatMessageAttachment.cs [hunk 12]
```diff
@@ -0,0 +1,138 @@
...
+    public sealed class SlackChatMessageAttachmentField
+    {
+        /// <summary>
+        /// Required Field Title
+        /// </summary>
+        /// <value>
+        /// The title.
+        /// </value>
+        public string Title { get; set; }
+        /// <summary>
```
Comment (devlead) [inline src/Cake.Slack/Chat/SlackChatMessageAttachment.cs]: Add new line between previous property and summary

File: src/Cake.Slack/Chat/SlackChatMessageAttachment.cs [hunk 13]
```diff
@@ -0,0 +1,138 @@
...
+        /// Text value of the field. May contain standard message markup and must be escaped as normal
+        /// </summary>
+        /// <value>
+        /// The value.
+        /// </value>
+        public string Value { get; set; }
+        /// <summary>
```
Comment (devlead) [inline src/Cake.Slack/Chat/SlackChatMessageAttachment.cs]: Add new line between previous property and summary

File: src/Cake.Slack/Chat/SlackChatMessageAttachment.cs [hunk 14]
```diff
@@ -0,0 +1,138 @@
...
+        /// The fields.
+        /// </value>
+        public IList<SlackChatMessageAttachmentField> Fields { get; set; }
+    }
+
+    /// <summary>
+    /// Field for message attachment
+    /// </summary>
+    [CakeAliasCategory("Slack")]
+    public sealed class SlackChatMessageAttachmentField
```
Comment (devlead) [inline src/Cake.Slack/Chat/SlackChatMessageAttachment.cs]: Move to own file `SlackChatMessageAttachmentField.cs`

**Comment by devlead** [comment]
Hi @RLittlesII,

Thanks for your contribution, made some minor code style comments.

After you fixed above Please rebase into 1 commit against master, then I'll test and merge if everything's ok. 

/M

**Comment by RLittlesII** [comment]
Rebase complete.

**Comment by devlead** [comment]
Thanks @RLittlesII! :+1: I'll get back to you as soon as I've tested this.

**Comment by devlead** [comment]
@RLittlesII merged, thanks for your contribution :+1:  I've also added support for attachments when using tokens.

**Pull request merged**